Libraries/libbom/Sources/bom.c

```diff
@@ -245,9 +255,16 @@ bom_index_add(struct bom_context *context, const void *data, size_t data_len)
     struct bom_index_header *index_header = (struct bom_index_header *)((void *)header + ntohl(header->index_offset));
 
     /* Insert index at the end of the list. */
+    bool resize = true;
+    if (context->preallocated_index_count > 0) {
```

Comment:
Sure, I could calculate out the new size of the index list and compare it to index_length in the bom header. But that is a lot more work compared to keeping track in a counter.

Also, the size calculation would have to compensate for those two extra slots that are needed empty at the end, which means keeping track of this special requirement in more than one place.